services/galaxytoolshed/galaxy-toolshed-base.js

```diff
@@ -0,0 +1,56 @@
+import Joi from 'joi'
+import { BaseJsonService } from '../index.js'
+
+const orderedInstallableRevisionsSchema = Joi.array()
+  .items(Joi.string())
+  .required()
```

Comment:
Is it ever possible for this to be an empty array? I suppose this could be analogous to a claimed package name that has no versions/revisions?

If it is possible, then we'll need to account for that empty array scenario in the code that handles the processing of the response. If it's not possible (or at least if we think it should be impossible), then we should update the schema:

```suggestion
const orderedInstallableRevisionsSchema = Joi.array()
  .items(Joi.string().required())
  .required()
```

Reply:
Yes you're right. This array could be empty due to undefined problems.